protoc-gen-swagger/genswagger/template.go

```diff
@@ -0,0 +1,432 @@
+	// /{my_field=prefix/*}/ and end up with 2 sections representing my_field.
+	// Instead do the right thing and write a small pushdown (counter) automata
+	// for it.
+	parts := []string{}
+	depth := 0
+	buffer := ""
+	for _, char := range path {
+		switch char {
+		case '{':
+			// Push on the stack
+			depth += 1
+			buffer += string(char)
+			break
+		case '}':
+			if depth == 0 {
+				panic("Encountered } without matching { before it.")
+			}
+			// Pop from the stack
+			depth -= 1
+			buffer += "}"
+		case '/':
+			if depth == 0 {
+				parts = append(parts, buffer)
+				buffer = ""
+				// Since the stack was empty when we hit the '/' we are done with this
+				// section.
+				continue
+			}
+		default:
+			buffer += string(char)
+			break
+		}
+	}
+
+	// Now append the last element to parts
+	parts = append(parts, buffer)
+
+	// Parts is now an array of segments of the path. Interestingly, since the
+	// syntax for this subsection CAN be handled by a regexp since it has no
+	// memory.
+	re := regexp.MustCompile("{([a-z][a-z0-9_]*).*}")
+	for index, part := range parts {
+		parts[index] = re.ReplaceAllString(part, "{$1}")
+	}
+
+	return strings.Join(parts, "/")
+}
+
+func renderServices(services []*descriptor.Service, paths swaggerPathsObject) error {
+	for _, svc := range services {
+		for _, meth := range svc.Methods {
+			if meth.GetClientStreaming() || meth.GetServerStreaming() {
+				return errors.New(`Service uses streaming, which is not currently supported. Maybe you would like to implement it? It wouldn't be that hard and we don't bite. Why don't you send a pull request to https://github.com/gengo/grpc-gateway?`)
+			}
+			for _, b := range meth.Bindings {
+				// Iterate over all the swagger parameters
+				parameters := swaggerParametersObject{}
+				for _, parameter := range b.PathParams {
+
+					var paramType, paramFormat string
+					switch parameter.Target.FieldDescriptorProto.Type.String() {
+					case "TYPE_DOUBLE":
+						paramType = "number"
+						paramFormat = "double"
+						break
+					case "TYPE_FLOAT":
+						paramType = "number"
+						paramFormat = "float"
+						break
+					case "TYPE_INT64":
+						paramType = "integer"
+						paramFormat = "int64"
+						break
+					case "TYPE_UINT64":
+						paramType = "integer"
+						paramFormat = "int64"
+						break
+					case "TYPE_INT32":
+						paramType = "integer"
+						paramFormat = "int32"
+						break
+					case "TYPE_FIXED64":
+						paramType = "integer"
+						paramFormat = "int64"
+						break
+					case "TYPE_FIXED32":
+						paramType = "integer"
+						paramFormat = "int32"
+						break
+					case "TYPE_BOOL":
+						paramType = "boolean"
+						paramFormat = "boolean"
+						break
+					case "TYPE_STRING":
+						paramType = "string"
+						paramFormat = "string"
+						break
+					case "TYPE_GROUP":
+						panic("Groups are not allowed in the path for a HTTP RPC. Please use a primitive type instead (string, int64, etc)")
```

Comment:
https://github.com/golang/go/wiki/CodeReviewComments#dont-panic

It is sufficient to return error even in this case.